### PR TITLE
feat(interaction): add `isRepliable` type guard

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

@almostSouji will be happy about this

**Status and versioning classification:**
Please move lines that apply to you out of the comment:
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/discord.js/src/structures/Interaction.js [hunk 1]
```diff
@@ -225,6 +225,15 @@ class Interaction extends Base {
...
     );
   }
+
+  /**
+   * Indicates whether this interaction can be replied to or not.
+   * @returns {boolean}
+   */
+  isRespondable() {
+    const thisType = InteractionType[this.type];
+    return thisType !== InteractionType.Ping || thisType !== InteractionType.ApplicationCommandAutocomplete;
```
Comment (almeidx) [inline packages/discord.js/src/structures/Interaction.js:235]:
```suggestion
    return thisType !== InteractionType.Ping && thisType !== InteractionType.ApplicationCommandAutocomplete;
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/Interaction.js (hunk 1 above)
Comment (suneettipirneni) [reply packages/discord.js/src/structures/Interaction.js]: You're always fixing my mistakes huh

File: packages/discord.js/src/structures/Interaction.js [hunk 2]
```diff
@@ -225,6 +225,15 @@ class Interaction extends Base {
...
     );
   }
+
+  /**
+   * Indicates whether this interaction can be replied to or not.
+   * @returns {boolean}
+   */
+  isRespondable() {
+    const thisType = InteractionType[this.type];
+    return thisType !== InteractionType.Ping && thisType !== InteractionType.ApplicationCommandAutocomplete;
```
Comment (ImRodry) [inline packages/discord.js/src/structures/Interaction.js:235]: shorter = better?
```suggestion
	return ![InteractionType.Ping, InteractionType.ApplicationCommandAutocomplete].includes(this.type)
```

Comment (almeidx) [reply packages/discord.js/src/structures/Interaction.js:235]: You'd have to either keep the `thisType` variable or use `InteractionType[this.type]` in the `includes()`

Comment (ImRodry) [reply packages/discord.js/src/structures/Interaction.js:235]: Yeah I messed that up, edited the comment now

**Comment by iCrawl** [comment]
@ImRodry injected some spacing issues, youd have to re-run prettier :P

File: packages/discord.js/src/structures/Interaction.js [hunk 3]
```diff
@@ -225,6 +225,14 @@ class Interaction extends Base {
       ComponentType[this.componentType] === ComponentType.SelectMenu
     );
   }
+
+  /**
+   * Indicates whether this interaction can be replied to or not.
+   * @returns {boolean}
+   */
+  isRespondable() {
```
Comment (SpaceEEC) [inline packages/discord.js/src/structures/Interaction.js:233]: Isn't the name slightly misleading as autocomplete interactions can be `respond`ed to?
Maybe `replyable` instead?

**Review by SpaceEEC — DISMISSED** [review]

File: packages/discord.js/src/structures/Interaction.js [hunk 4]
```diff
@@ -225,6 +225,14 @@ class Interaction extends Base {
       ComponentType[this.componentType] === ComponentType.SelectMenu
     );
   }
+
+  /**
+   * Indicates whether this interaction can be replied to or not.
```
Comment (almeidx) [inline packages/discord.js/src/structures/Interaction.js:230]:
```suggestion
   * Indicates whether this interaction can be replied to.
```

**Comment by imranbarbhuiya** [comment]
As interaction can be replied only once so `isRepliable` seems like can u still reply to the interaction, or it's already replied/deferred.

File: packages/discord.js/typings/index.d.ts [hunk 5]
```diff
@@ -320,6 +320,17 @@ export type GuildCacheMessage<Cached extends CacheType> = CacheTypeReducer<
   Message | APIMessage
 >;
 
+export interface InteractionResponses<Cached extends CacheType> {
```
Comment (kyranet) [inline packages/discord.js/typings/index.d.ts:323]: I'm not a fan of this name considering it's in plural. Also, doesn't it remove the other properties the object may have?

Comment (kyranet) [reply packages/discord.js/typings/index.d.ts:323]: By the way, this might need type tests so we can ensure the type is narrowed correctly.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Comment by iCrawl** [comment]
Needs a rebase.

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**